### ISSUE TITLE
OpenSSL3 required on macOS and homebrew?

### ISSUE THREAD

**Issue by psulightning** [brianmario/mysql2#1300]
I tried to build mysql2 0.5.4 on macOS Ventura.  Ran into similar problem with OpenSSL as many past issues.
```
ld: warning: directory not found for option '-L/usr/local/opt/openssl@3/lib'
ld: library not found for -lssl
```
In my bundle config for mysql2, I have `--with-opt-dir=$(brew --prefix openssl@1.1) --with-mysql-config=/usr/local/opt/mysql@5.7/bin/mysql_config`.  Naturally, it confused me as to why the build error references openssl@3.  Then I noticed in extconf.rb this block: 

https://github.com/brianmario/mysql2/blob/bab4969264250abca836e85a9c332c42be573dfc/ext/mysql2/extconf.rb#L30-L33

By using `brew --prefix openssl`, this causes `make` to use openssl@3 even if not installed and ignores the "--with-opt-dir" parameter.

Is this intended for mysql2 on macOS going forward?  Should this get removed in favor of the older workaround with "--with-opt-dir"

**Comment by sodabrew** [comment]
The user input should always override an auto-detected value, so I would consider this a bug.

I think it's coming up now because Homebrew is assuming that `openssl` means `openssl@3` even if `openssl@1.1` is installed. I would like to know if Homebrew considers this a bug and will fix it, or if there's some expected change on the part of consuming software to do something differently.

**Comment by junaruga** [comment]
You can see https://github.com/Homebrew/brew/issues/12879#issuecomment-1039613762 . The brew openssl package maintainer said it's intentional.

**Comment by psulightning** [comment]
I accept that @junaruga from Homebrew's perspective.  The question really is should mysql2 ignore the user input.  As @sodabrew indicated, it should not.

**Comment by junaruga** [comment]
I opened the PR https://github.com/brianmario/mysql2/pull/1301 to add the custom Homebrew openssl directory configuration option.

**Comment by sodabrew** [comment]
Changes landed with #1303, will be in next release! If `--with-opt-dir` is specified but invalid it's ignored. The new option `--with-openssl-dir` checks that it's given a valid path.

**Comment by sodabrew** [comment]
Closing - Further improved docs in #1306
